### PR TITLE
Aabb intervals

### PR DESCRIPTION
**Pull request by hollasch**
Axis-Aligned Bounding Box class now uses intervals for each axis.

### CODE REVIEW

**Comment by hollasch** [comment]
This change also addresses #773

File: books/RayTracingTheNextWeek.html [hunk 1]
```diff
@@ -1024,7 +1037,7 @@
         if (!a->bounding_box(0,0, box_a) || !b->bounding_box(0,0, box_b))
             std::cerr << "No bounding box in bvh_node constructor.\n";
 
-        return box_a.min().e[axis] < box_b.min().e[axis];
+        return box_a.axis(axis).min < box_b.axis(axis).min;
```
Comment (trevordblack) [inline books/RayTracingTheNextWeek.html:1040]: maybe make it more explicit that `axis()` is the function that returns an interval, and `axis` is an int tells that function what to return.
Maybe change the `axis` variable to `a` or `axis_int`

File: src/TheNextWeek/bvh.h [hunk 2]
```diff
@@ -49,7 +49,7 @@ inline bool box_compare(const shared_ptr<hittable> a, const shared_ptr<hittable>
     if (!a->bounding_box(0,0, box_a) || !b->bounding_box(0,0, box_b))
         std::cerr << "No bounding box in bvh_node constructor.\n";
 
-    return box_a.min().e[axis] < box_b.min().e[axis];
+    return box_a.axis(axis).min < box_b.axis(axis).min;
```
Comment (trevordblack) [inline src/TheNextWeek/bvh.h:52]: See comment above

File: src/TheRestOfYourLife/bvh.h [hunk 3]
```diff
@@ -49,7 +49,7 @@ inline bool box_compare(const shared_ptr<hittable> a, const shared_ptr<hittable>
     if (!a->bounding_box(0,0, box_a) || !b->bounding_box(0,0, box_b))
         std::cerr << "No bounding box in bvh_node constructor.\n";
 
-    return box_a.min().e[axis] < box_b.min().e[axis];
+    return box_a.axis(axis).min < box_b.axis(axis).min;
```
Comment (trevordblack) [inline src/TheRestOfYourLife/bvh.h:52]: see comments above

**Review by trevordblack — APPROVED** [review]

File: books/RayTracingTheNextWeek.html (hunk 1 above)
Comment (hollasch) [reply books/RayTracingTheNextWeek.html:1040]: Yeah, I can do better than what I've got.

**Pull request merged**